### PR TITLE
ADD: try-runtime relative config

### PR DESCRIPTION
**Pull request by coderfengyun**
Support try-runtime for parami blockchain

* for test before on-runtime-upgrade

**Type of change**

* [ ] Bug Fix (non-breaking change which fixes an issue)
* [x] New Feature (non-breaking change which adds functionality)
* [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update
* [ ] This change requires a storage migration
* [ ] This change requires a benchmark

**Checklist**

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] New and existing benchmarks pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream modules
* [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pallets/magic/Cargo.toml [hunk 1]
```diff
@@ -27,6 +27,7 @@ parami-did = { path = '../did', default-features = false }
 parami-traits = { path = '../traits', default-features = false }
 
 serde = { version = '1.0', optional = true }
+log = { version = "0.4.14", default-features = false }
```
Comment (kinosang) [inline pallets/magic/Cargo.toml:30]: consider using `tracing` cuz substrate use it.

File: runtimes/dana/Cargo.toml [hunk 2]
```diff
@@ -30,6 +30,7 @@ version = '1.0'
 [dependencies]
 hex-literal = { version = '0.3', optional = true }
 serde = { version = '1.0', optional = true, features = ['derive'] }
+log = { version = "0.4.14", default-features = false }
```
Comment (kinosang) [inline runtimes/dana/Cargo.toml:33]: `tracing` too

File: runtimes/dana/Cargo.toml [hunk 3]
```diff
@@ -236,3 +240,44 @@ runtime-benchmarks = [
...
+    'pallet-offences/try-runtime',
+    'pallet-scheduler/try-runtime',
+    'pallet-society/try-runtime',
+    'pallet-staking/try-runtime',
+    'pallet-timestamp/try-runtime',
+    'pallet-tips/try-runtime',
+    'pallet-treasury/try-runtime',
+    'pallet-uniques/try-runtime',
+    'pallet-utility/try-runtime',
+    'pallet-vesting/try-runtime',
```
Comment (kinosang) [inline runtimes/dana/Cargo.toml:282]: we might not need to do test for substrate frames

File: runtimes/para/Cargo.toml [hunk 4]
```diff
@@ -31,6 +31,7 @@ version = '1.0'
 hex-literal = { version = '0.3', optional = true }
 serde = { version = '1.0', optional = true, features = ['derive'] }
 smallvec = '1.6'
+log = { version = "0.4.14", default-features = false }
```
Comment (kinosang) [inline runtimes/para/Cargo.toml:34]: `tracing` too

File: runtimes/dana/Cargo.toml (hunk 2 above)
Comment (coderfengyun) [reply runtimes/dana/Cargo.toml:33]: When I check substrate's code, they just add this log dependency, we don't need to simplify it anymore ?

File: runtimes/para/Cargo.toml (hunk 4 above)
Comment (coderfengyun) [reply runtimes/para/Cargo.toml:34]: Same with above.

File: runtimes/para/Cargo.toml [hunk 5]
```diff
@@ -278,3 +283,42 @@ runtime-benchmarks = [
...
+    'pallet-offences/try-runtime',
+    'pallet-scheduler/try-runtime',
+    'pallet-society/try-runtime',
+    'pallet-staking/try-runtime',
+    'pallet-timestamp/try-runtime',
+    'pallet-tips/try-runtime',
+    'pallet-treasury/try-runtime',
+    'pallet-uniques/try-runtime',
+    'pallet-utility/try-runtime',
+    'pallet-vesting/try-runtime',
```
[the remark this answers is not in the document]
Comment (coderfengyun) [reply runtimes/para/Cargo.toml:323]: Same with above.

File: runtimes/dana/Cargo.toml (hunk 2 above)
Comment (kinosang) [reply runtimes/dana/Cargo.toml:33]: @coderfengyun seems they use `log` in cli and runtime and `tracing` in frames (pallets).

**Review by kinosang — APPROVED** [review]
LGFM

**Pull request merged**